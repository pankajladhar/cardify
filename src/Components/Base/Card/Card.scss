.Card {
  color: rgba(0, 0, 0, 0.87);
  background-color: rgb(255, 255, 255);
  transition: all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms;
  box-sizing: border-box;
  -webkit-tap-highlight-color: rgba(0, 0, 0, 0);
  box-shadow: rgba(0, 0, 0, 0.12) 0px 1px 6px, rgba(0, 0, 0, 0.12) 0px 1px 4px;
  border-radius: 2px;
  padding: 12px;
  display: flex;
  align-items: center;
  color: #fff;


  .Card__Info {
    display: flex;
    flex-direction: column;
  }

  .Card__Info__Main {
    color: #000;
    font-size: 16px;
    line-height: 1;
  }

  .Card__Info__Main__Alias {
    color: rgba(0, 0, 0, .54);
    font-size: 14px;
    line-height: 1;
    padding-top: 5px;
  }
}

.Card__Avatar {
  width: 40px;
  height: 40px;
  border-radius:50%;
  margin-right: 12px;
  display: flex;
  align-items: center;
  justify-content: center;
  font-size: 20px;

  &.Card__Avatar--Tale {
    background-color: #009688;
  }

  &.Card__Avatar--Purple {
    background-color: #9C27B0;
  }

  &.Card__Avatar--Indigo {
    background-color: #3F51B5;
  }

  &.Card__Avatar--Pink {
    background-color: #E91E63;
  }

  &.Card__Avatar--DeepOrange {
    background-color: #FF5722;
  }
}
