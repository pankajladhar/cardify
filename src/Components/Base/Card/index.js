import React, { Component } from 'react';
import PropTypes from 'prop-types'

class Card extends Component {
    constructor(props) {
        super(props);
    }

    render() {
        const { cardType, alias } = this.props;
        return(
            <div className="Card">
                { cardType } { alias }
            </div>
        )
    }
}

Card.propTypes = {
    id: PropTypes.string.isRequired,
    cardType: PropTypes.string.isRequired,
    alias: PropTypes.string,
}

export default Card;