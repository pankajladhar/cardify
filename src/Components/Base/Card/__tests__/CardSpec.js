import React from 'react';
import renderer from 'react-test-renderer';
import Card from './../index';

describe('Card', () => {
    let cardProps = {}
    
    beforeEach(()=>{
        cardProps = {
            id : "122df32q",
            cardType: "HDFC"
        }
    })

    it('should render correctly with all mandatory props', () =>{
        let tree = renderer.create(
            <Card {...cardProps}/>
        ).toJSON();

        expect(tree).toMatchSnapshot()
    });

    it('should render correctly with all props', ()=>{
        cardProps.alias = "HDFC alias";
        let tree = renderer.create(
            <Card {...cardProps} />
        ).toJSON();
        expect(tree).toMatchSnapshot();
    })
})