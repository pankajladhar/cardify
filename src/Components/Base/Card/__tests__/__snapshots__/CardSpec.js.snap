// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should render correctly with all mandatory props 1`] = `
<div
  className="Card"
>
  HDFC
   
</div>
`;

exports[`Card should render correctly with all props 1`] = `
<div
  className="Card"
>
  HDFC
   
  HDFC alias
</div>
`;
